### PR TITLE
qjoypad: 4.1.0 -> 4.3.0

### PR DESCRIPTION
**Pull request by radraw**
###### Motivation for this change
Switching to Qt5 makes the package play well with modern desktop environments. Tested on Ubuntu 18.04, default Ubuntu session.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @astsmtl

### CODE REVIEW

File: pkgs/tools/misc/qjoypad/default.nix [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-    sha256 = "1jlm7i26nfp185xrl41kz5z6fgvyj51bjpz48cg27xx64y40iamm";
+  name = "qjoypad-${version}";
+  version = "4.3.0";
+  
+  src = fetchFromGitHub {
+    owner = "panzi";
+    repo = "qjoypad";
+    rev = "v${version}";
+    sha256 = "0ps85ncqmy8jz3zzyv5f5z4564kkgcl799ri4kcz407mjyc4fvs6";
   };
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
  };
  
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ libX11 libXtst qt4 ];
+  nativeBuildInputs = [ pkgconfig cmake ];
+  buildInputs = [ libX11 libXtst qt5.qtbase qt5.qttools qt5.qtx11extras ];
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
  buildInputs = [ libX11 libXtst qtbase qttools qtx11extras ];
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
   NIX_LDFLAGS = [ "-lX11" ];
-  patchPhase = ''
-    cd src
-    substituteInPlace config --replace /bin/bash ${stdenv.shell}
-    mkdir -p $out
-    export NIX_LDFLAGS="$NIX_LDFLAGS -rpath ${libX11}/lib"
-  '';
   meta = {
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]: personal preference, but also very common
```suggestion
  meta = with lib {
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 4]
```diff
@@ -32,7 +31,7 @@ stdenv.mkDerivation rec {
       of gaming devices in Linux, and makes the Linux gaming
       experience just a little bit nicer.
     '';
-    homepage = http://qjoypad.sourceforge.net;
+    homepage = https://github.com/panzi/qjoypad;
     license = stdenv.lib.licenses.gpl2;
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
    license = licenses.gpl2;
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 5]
```diff
@@ -32,7 +31,7 @@ stdenv.mkDerivation rec {
...
     maintainers = with stdenv.lib.maintainers; [ astsmtl ];
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
    maintainers = with maintainers; [ astsmtl ];
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 6]
```diff
@@ -32,7 +31,7 @@ stdenv.mkDerivation rec {
...
     platforms = with stdenv.lib.platforms; linux;
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
    platforms = with platforms; linux;
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 7]
```diff
@@ -1,19 +1,18 @@
-{ stdenv, fetchurl, pkgconfig, libX11, libXtst, qt4 }:
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, libX11, libXtst, qt5 }:
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
{ lib, fetchFromGitHub, cmake, pkgconfig, libX11, libXtst, qtbase, qttools, qtx11extras }:
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 8]
```diff
@@ -1,19 +1,18 @@
...
+
 stdenv.mkDerivation rec {
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
mkDerivation rec {
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 9]
```diff
@@ -1,19 +1,18 @@
...
-  name = "qjoypad-4.1.0";
-  src = fetchurl {
-    url = "mirror://sourceforge/qjoypad/${name}.tar.gz";
-    sha256 = "1jlm7i26nfp185xrl41kz5z6fgvyj51bjpz48cg27xx64y40iamm";
+  name = "qjoypad-${version}";
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix]:
```suggestion
  pname = "qjoypad";
```

**Review by jonringer — COMMENTED** [review]
running it as is creates:
```
[nix-shell:/home/jon/.cache/nix-review/pr-74330-1]$ ./results/qjoypad/bin/qjoypad
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```

please read: https://nixos.org/nixpkgs/manual/#sec-language-qt

**Comment by jonringer** [comment]
dont forget to squash the commits

```
git reset HEAD~8
git add pkgs/
git commit --amend --no-edit
git push <fork> <branch> --force
```

**Comment by radraw** [comment]
Now when I run nix-build, I get this error:
error: syntax error, unexpected '}', at /home/radek/proj/nixpkgs/pkgs/tools/misc/qjoypad/default.nix:40:1

**Review by jonringer — COMMENTED** [review]
please squash all the commits to adhere to contributing.md

```
git reset HEAD~4
git add pkgs/
git commit --amend --no-edit
git push <fork> <branch> --force
```

**Comment by radraw** [comment]
Done, thank you for your help !

**Comment by jonringer** [comment]
please resolve conflict, otherwise it LGTM

**Comment by drewrisinger** [comment]
ping @radraw for resolving merge conflict

**Comment by radraw** [comment]
I see the change hasn't been picked yet. Is something missing I can help to proceed with it ?

File: pkgs/tools/misc/qjoypad/default.nix [hunk 10]
```diff
@@ -32,9 +32,9 @@ stdenv.mkDerivation rec {
       of gaming devices in Linux, and makes the Linux gaming
       experience just a little bit nicer.
     '';
-    homepage = "http://qjoypad.sourceforge.net";
-    license = stdenv.lib.licenses.gpl2;
-    maintainers = with stdenv.lib.maintainers; [ astsmtl ];
-    platforms = with stdenv.lib.platforms; linux;
+    homepage = https://github.com/panzi/qjoypad;
```
Comment (jonringer) [inline pkgs/tools/misc/qjoypad/default.nix:35]: rfc 45
```suggestion
    homepage = "https://github.com/panzi/qjoypad";
```

**Review by jonringer — COMMENTED** [review]
to have clean git histories, please rebase on top of the target branch:

```
git pull -r origin master
```

instead of merging upstream into your branch :)
should just be 1 commit

File: pkgs/tools/misc/qjoypad/default.nix [hunk 11]
```diff
@@ -1,20 +1,20 @@
-{ lib, stdenv, fetchurl, pkg-config, libX11, libXtst, qt4 }:
-stdenv.mkDerivation rec {
-  name = "qjoypad-4.1.0";
-  src = fetchurl {
-    url = "mirror://sourceforge/qjoypad/${name}.tar.gz";
-    sha256 = "1jlm7i26nfp185xrl41kz5z6fgvyj51bjpz48cg27xx64y40iamm";
+{ lib, mkDerivation, fetchFromGitHub, cmake, pkgconfig, libX11, libXtst, qtbase, qttools, qtx11extras }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/qjoypad/default.nix:1]:
```suggestion
{ lib, mkDerivation, fetchFromGitHub, cmake, pkg-config, libX11, libXtst, qtbase, qttools, qtx11extras }:
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 12]
```diff
@@ -1,20 +1,20 @@
...
-  NIX_LDFLAGS = "-lX11";
-  patchPhase = ''
-    cd src
-    substituteInPlace config --replace /bin/bash ${stdenv.shell}
-    mkdir -p $out
-    export NIX_LDFLAGS="$NIX_LDFLAGS -rpath ${libX11}/lib"
-  '';
-  meta = {
+
+  nativeBuildInputs = [ pkgconfig cmake ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/qjoypad/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config cmake ];
```

File: pkgs/tools/misc/qjoypad/default.nix [hunk 13]
```diff
@@ -32,9 +32,9 @@ stdenv.mkDerivation rec {
       of gaming devices in Linux, and makes the Linux gaming
       experience just a little bit nicer.
     '';
-    homepage = "http://qjoypad.sourceforge.net";
-    license = lib.licenses.gpl2;
-    maintainers = with lib.maintainers; [ astsmtl ];
-    platforms = with lib.platforms; linux;
+    homepage = "https://github.com/panzi/qjoypad";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/qjoypad/default.nix:36]: gpl2Only or gpl2Plus?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/qjoypad/default.nix (hunk 13 above)
Comment (radraw) [reply pkgs/tools/misc/qjoypad/default.nix:36]: This project has GPL 2.0 license as I see, so I would assume GPL 2.0 only.

**Pull request closed without merging**